### PR TITLE
Allow run and connect container onto swarm networks

### PR DESCRIPTION
**Pull request by aboch**
This PR proposes one possible solution to allow users to run (and connect) containers to swarm overlay networks on any swarm node.

While it still depends on non merged/reviewed changes in other projects, and need some tweaks , I wanted to open it so that the e2e change can be seen and tried by others.

In order to run container on a swarm network, the network must be created with a new `--legacy` flag.

```
$ docker swarm init
Swarm initialized: current node (2g1d5i7w8a9dl50769dnyxs9u) is now a manager.

<snip>

$ 
$ docker network create -d overlay --legacy ovl
at79w26ljvzpxupvrkie0s3of
$ docker run --rm --network ovl busybox ifconfig
eth0      Link encap:Ethernet  HWaddr 02:42:0A:00:00:02  
          inet addr:10.0.0.2  Bcast:0.0.0.0  Mask:255.255.255.0
          inet6 addr: fe80::42:aff:fe00:2/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1450  Metric:1
          RX packets:1 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:90 (90.0 B)  TX bytes:90 (90.0 B)

eth1      Link encap:Ethernet  HWaddr 02:42:AC:13:00:02  
          inet addr:172.19.0.2  Bcast:0.0.0.0  Mask:255.255.0.0
          inet6 addr: fe80::42:acff:fe13:2/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:1 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:90 (90.0 B)  TX bytes:90 (90.0 B)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

$
```

### CODE REVIEW

File: api/client/network/create.go [hunk 1]
```diff
@@ -55,6 +56,7 @@ func newCreateCommand(dockerCli *client.DockerCli) *cobra.Command {
 	flags.StringSliceVar(&opts.labels, "label", []string{}, "Set metadata on a network")
 	flags.BoolVar(&opts.internal, "internal", false, "Restrict external access to the network")
 	flags.BoolVar(&opts.ipv6, "ipv6", false, "Enable IPv6 networking")
+	flags.BoolVar(&opts.legacy, "legacy", false, "Allows run container on legacy swarm network")
```
Comment (justincormack) [inline api/client/network/create.go]: Do you mean "Allows running legacy container on swarm network"?

**Comment by justincormack** [comment]
Ok this is working for me. I needed it because I need to run some containers with features that are not yet available for services (privileges and devices) but they still want to talk to the rest of the swarm.

**Comment by justincormack** [comment]
After a while using this I get `docker: Error response from daemon: timed out while waiting for endpoint IPAM allocation.` when I try to run containers on my legacy network.

**Comment by cpuguy83** [comment]
Closed the engine-api PR which adds the `Legacy` field.
As mentioned, `--legacy` doesn't really mean anything. What happens in another year when we don't like what we currently have and we have yet another legacy?

This should be able to work without requiring the user to set an ambiguous field.

**Comment by aboch** [comment]
Closing in favor of #25962

**Pull request closed without merging**